packages/cli/src/options/beaconNodeOptions/builder.ts

```diff
@@ -56,4 +59,12 @@ export const options: CliCommandOptions<ExecutionBuilderArgs> = {
     description: "Number of missed slots allowed in the faultInspectionWindow for builder circuit",
     group: "builder",
   },
+
+  "builder.userAgent": {
```

Comment:
is this a value we want operators to mess with? defeats the whole purpose of what this PR is trying to achieve. Maybe hiding it would be better

Reply:
Given that this leaks information I prefer to expose the option to mess with it. Either:
- A flag to prevent sending this information
- A flag to control what is sent on the user agent
We should do something similar for the p2p agent

Reply:
I like `--private` flag (or similar) suggested in https://github.com/ChainSafe/lodestar/issues/2438, sounds much more elegant option for users with privacy/security concerns, can of course still keep flags to individually override each of those values

Reply:
Feel free to PR on top of unstable